### ISSUE TITLE
Merger?

### ISSUE THREAD

**Issue by icls1337** [intel/llvm#7152]
https://github.com/0xADE1A1DE/USLH
https://ia.cr/2022/715
In this paper we revisit the Spectre v1 vulnerability and software-only countermeasures. Specifically, we systematically investigate the performance penalty and security properties of multiple variants of speculative load hardening (SLH). As part of this investigation we implement the “strong SLH” variant by Patrignani and Guarnieri (CCS 2021) as a compiler extension to LLVM. We show that none of the existing variants, including strong SLH, is able to protect against all Spectre v1 attacks in practice. We do this by demonstrating, for the first time, that variable-time arithmetic instructions leak secret information even if they are executed only speculatively. We extend strong SLH to include protections also against this kind of leakage, implement the resulting full protection in LLVM, and use the SPEC2017 benchmarks to compare its performance to the existing variants of SLH and to code that uses fencing instructions to completely prevent speculative execution. We show that our proposed countermeasure is able to offer full protection against Spectre v1 attacks at much better performance than code using fences. In fact, for several benchmarks our approach is more than twice as fast.

**Comment by lvwr** [comment]
Hi, thank you for the report and for the pointers. Please, allow me some time to read through and take a better look. Also, please, keep us posted about any further developments.

**Comment by AlexeySachkov** [comment]
I agree with @0x12CC here. This repo is a fork of the upstream LLVM to add SYCL implementation to it. Core LLVM is being developed in upstream and the paper doesn't seem to have anything to do with SYCL.

I'm going to close the issue, but please speak up if you disagree or have some comments.
